packages/amplify-util-mock/src/CFNParser/stack/index.ts

```diff
@@ -87,22 +87,24 @@ export function getDependencyResources(node: object | any[], params: Record<stri
     return [];
   }
 
-  if (isPlainObject(node) && Object.keys(node).length === 1) {
-    const fnName = Object.keys(node)[0];
-    const fnArgs = node[fnName];
-    if ('Ref' === fnName) {
-      const resourceName = fnArgs;
-      if (!Object.keys(params).includes(resourceName)) {
+  if (isPlainObject(node)) {
+    const nodeKeys = Object.keys(node);
+    if (nodeKeys.length === 1) {
+      const fnName = Object.keys(node)[0];
+      const fnArgs = node[fnName];
+      if ('Ref' === fnName) {
+        const resourceName = fnArgs;
+        if (!Object.keys(params).includes(resourceName)) {
+          result.push(resourceName);
+          return result;
+        }
+      } else if ('Fn::GetAtt' === fnName) {
```

Comment:
nit: consider moving the variable to left and constant to right for readability.